org.eclipse.lsp4e.jdt/src/org/eclipse/lsp4e/jdt/LSJavaProposal.java

```diff
@@ -20,10 +21,22 @@
 import org.eclipse.swt.graphics.Image;
 import org.eclipse.swt.graphics.Point;
 
+@SuppressWarnings("restriction")
 class LSJavaProposal implements IJavaCompletionProposal {
+	
+	private static final int LS_DEFAULT_RELEVANCE = 18;
+	
+	private static final int MAX_BASE_RELEVANCE = 51 * 16; // Looks like JDT's max for exact match is 52
+	
+	 // Based on org.eclipse.jdt.internal.ui.text.java.RelevanceComputer
+	private static final int DEFAULT_RELEVANCE = (RelevanceConstants.R_DEFAULT + LS_DEFAULT_RELEVANCE) * 16;
 
-	protected ICompletionProposal delegate;
+	private static final int RANGE_WITHIN_CATEGORY = Math.round((MAX_BASE_RELEVANCE - DEFAULT_RELEVANCE) / 4);
```

Comment:
This is either suboptimal or wrong:
```java
private static final int RANGE_WITHIN_CATEGORY = Math.round((MAX_BASE_RELEVANCE - DEFAULT_RELEVANCE) / 4);
```
because `(MAX_BASE_RELEVANCE - DEFAULT_RELEVANCE) / 4`  returns an int and not a float.

So it should either be:
```java
//  removed the Math.round() no-op;
private static final int RANGE_WITHIN_CATEGORY = (MAX_BASE_RELEVANCE - DEFAULT_RELEVANCE) / 4;
`````
or:
```java
// cast dividend to float
private static final int RANGE_WITHIN_CATEGORY = Math.round((float) (MAX_BASE_RELEVANCE - DEFAULT_RELEVANCE) / 4);
```
or:
```java
// use a float divisor
private static final int RANGE_WITHIN_CATEGORY = Math.round((MAX_BASE_RELEVANCE - DEFAULT_RELEVANCE) / 4f);
```

depending on what the intention is.